stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel_actions/ChannelMembersAdapter.kt

```diff
@@ -0,0 +1,50 @@
+package io.getstream.chat.android.ui.channel_actions
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import io.getstream.chat.android.client.models.Member
+import io.getstream.chat.android.client.models.name
+import io.getstream.chat.android.ui.databinding.StreamItemChannelMemberBinding
+
+internal class ChannelMembersAdapter(
+    private val onMemberClicked: (Member) -> Unit
+) : ListAdapter<Member, ChannelMemberViewHolder>(
+    object : DiffUtil.ItemCallback<Member>() {
+        override fun areItemsTheSame(oldItem: Member, newItem: Member): Boolean {
+            return oldItem.user.id == newItem.user.id
+        }
+
+        override fun areContentsTheSame(oldItem: Member, newItem: Member): Boolean {
+            return oldItem == newItem
+        }
+    }
+) {
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ChannelMemberViewHolder {
+        return StreamItemChannelMemberBinding
+            .inflate(LayoutInflater.from(parent.context), parent, false)
+            .let { ChannelMemberViewHolder(it, onMemberClicked) }
+    }
+
+    override fun onBindViewHolder(holder: ChannelMemberViewHolder, position: Int) {
+        return holder.bind(getItem(position))
+    }
+}
+
+internal class ChannelMemberViewHolder(
```

Comment:
I'd nest this ViewHolder within the adapter as well, since it's only used within that class. Reduces its scope a bit.

Reply:
Done